Modules/Devices/src/Devices.Domain/Entities/Identities/IdentityDeletionProcess.cs

```diff
@@ -120,10 +122,30 @@ public void GracePeriodReminder3Sent(IdentityAddress address)
 
     public void Approve(IdentityAddress address, DeviceId approvedByDevice)
     {
-        if (Status != DeletionProcessStatus.WaitingForApproval)
-            throw new DomainException(DomainErrors.NoDeletionProcessWithRequiredStatusExists());
+        EnsureDeletionProcessIsInWaitingForApprovalStatus();
 
         Approve(approvedByDevice);
         _auditLog.Add(IdentityDeletionProcessAuditLogEntry.ProcessApproved(Id, address, approvedByDevice));
     }
+
+    public void Reject(IdentityAddress address, DeviceId rejectedByDevice)
+    {
+        EnsureDeletionProcessIsInWaitingForApprovalStatus();
+
+        Reject(rejectedByDevice);
+        _auditLog.Add(IdentityDeletionProcessAuditLogEntry.ProcessRejected(Id, address, rejectedByDevice));
+    }
+
+    private void EnsureDeletionProcessIsInWaitingForApprovalStatus()
+    {
+        if (Status != DeletionProcessStatus.WaitingForApproval)
+            throw new DomainException(DomainErrors.NoDeletionProcessWithRequiredStatusExists());
```

Comment:
We could improve the error message by passing the expected status. Then something like "The expected status is '{expectedStatus}'" could be added to the error message.